### ISSUE TITLE
Cannot install Desktop version of Github (64bit) (Windows 11)

### ISSUE THREAD

**Issue by EthanGitHub13** [desktop/desktop#18865]
### The problem

Cannot install Github

### Release version

June 2024 3.4.1 

### Operating system

Windows 11

### Steps to reproduce the behavior

After downloading the Github desktop setup file, I get an error of "Installation has failed"

### Log files

19200> 2022-07-26 23:28:15> Program: Starting Squirrel Updater: --install .
19200> 2022-07-26 23:28:15> Program: Starting install, writing to C:\Users\ekwei\AppData\Local\SquirrelTemp
19200> 2022-07-26 23:28:15> Program: About to install to: C:\Users\ekwei\AppData\Local\Discord
19200> 2022-07-26 23:28:15> SingleGlobalInstance: Grabbing lockfile with timeout of 00:00:10
19200> 2022-07-26 23:28:15> CheckForUpdateImpl: Reading RELEASES file from C:\Users\ekwei\AppData\Local\SquirrelTemp
19200> 2022-07-26 23:28:15> CheckForUpdateImpl: Remote version 1.0.9005 differs from local 
19200> 2022-07-26 23:28:15> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
19200> 2022-07-26 23:28:15> ApplyReleasesImpl: Writing files to app directory: C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\app.ico to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\app.ico
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\chrome_200_percent.pak to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\chrome_200_percent.pak
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\d3dcompiler_47.dll to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\d3dcompiler_47.dll
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\chrome_100_percent.pak to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\chrome_100_percent.pak
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\Discord.exe to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\Discord.exe
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\ffmpeg.dll to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\ffmpeg.dll
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\icudtl.dat to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\icudtl.dat
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\installer.db to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\installer.db
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\libEGL.dll to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\libEGL.dll
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\libGLESv2.dll to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\libGLESv2.dll
19200> 2022-07-26 23:28:16> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\resources.pak to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\resources.pak
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\snapshot_blob.bin to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\snapshot_blob.bin
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\updater.node to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\updater.node
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\vk_swiftshader.dll to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\vk_swiftshader.dll
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\vulkan-1.dll to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\vulkan-1.dll
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\v8_context_snapshot.bin to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\v8_context_snapshot.bin
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Moving file C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\lib\net45\vk_swiftshader_icd.json to C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\vk_swiftshader_icd.json
19200> 2022-07-26 23:28:17> ApplyReleasesImpl: Squirrel Enabled Apps: [C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\Discord.exe]
19200> 2022-07-26 23:28:17> Utility: Process Started: C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\Discord.exe --squirrel-install 1.0.9005, pid 7040
19200> 2022-07-26 23:28:18> Utility: Received exitcode 0 from process C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\Discord.exe
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: ran C:\Users\ekwei\AppData\Local\Discord\app-1.0.9005\Discord.exe, pid 1016
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: Starting fixPinnedExecutables
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: fixPinnedExecutables: newCurrentFolder: app-1.0.9005
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: File 'C:\Users\ekwei\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Accessibility\Narrator.lnk' could not be converted into a valid ShellLink: System.IO.FileNotFoundException: The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
   at Squirrel.Shell.ShellLink.IShellLinkW.Resolve(IntPtr hWnd, UInt32 fFlags)
   at Squirrel.Shell.ShellLink.Open(String linkFile, IntPtr hWnd, EShellLinkResolveFlags resolveFlags, UInt16 timeOut)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<fixPinnedExecutables>b__11_0(FileInfo file)
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: File 'C:\Users\ekwei\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Accessibility\On-Screen Keyboard.lnk' could not be converted into a valid ShellLink: System.IO.FileNotFoundException: The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
   at Squirrel.Shell.ShellLink.IShellLinkW.Resolve(IntPtr hWnd, UInt32 fFlags)
   at Squirrel.Shell.ShellLink.Open(String linkFile, IntPtr hWnd, EShellLinkResolveFlags resolveFlags, UInt16 timeOut)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<fixPinnedExecutables>b__11_0(FileInfo file)
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: Updating shortcut C:\Users\ekwei\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Discord Inc\Discord.lnk
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: Updating shortcut C:\Users\ekwei\OneDrive\Desktop\Discord.lnk
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: Fixing up tray icons
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: cleanDeadVersions: for version 1.0.9005
19200> 2022-07-26 23:28:18> ApplyReleasesImpl: cleanDeadVersions: exclude current version folder app-1.0.9005

**Comment by niik** [comment]
Hey @EthanGitHub13, thanks for reaching out. Based on the log file you provided it seems you're trying to install Discord and not GitHub Desktop. Please download GitHub Desktop from https://desktop.github.com and give it another go. If that fails please upload a new log file for that installation only.

Best of luck!

**Comment by EthanGitHub13** [comment]
> Hey @EthanGitHub13, thanks for reaching out. Based on the log file you provided it seems you're trying to install Discord and not GitHub Desktop. Please download GitHub Desktop from https://desktop.github.com and give it another go. If that fails please upload a new log file for that installation only.
> 
> Best of luck!

@niik The thing is that I am trying to install Github Desktop and I used that link before (and I just tried again as of the time of writing this) but it does not install Github Desktop and keeps prompting the same error when I click on it to activate the setup file. I heard it is because both discord and github desktop use the same squirrel installer. I found a very similar issue from a couple years ago that is like mine but unresolved. https://github.com/desktop/desktop/issues/5884

**Comment by niik** [comment]
@EthanGitHub13 The issue you linked to suggests fully uninstalling both GitHub Desktop and Discord (and deleting both of their directories inside of %LOCALAPPDATA% before reinstalling. Have you tried that?

There's not much else I can do to help here. The log file you provided definitely shows Discord trying to be installed and not Desktop so either you're running the wrong installer or there's something very interesting going on with your environment.

**Comment by espaulding** [comment]
I've run into this exact same issue with a fairly fresh windows 11 install. 

There appears to be an underlying collision between the installer for GitHub desktop and Discord.
If a user installs discord first they will have this issue when trying to install GitHub desktop

I did the following
1) fully uninstalled discord
2) deleted the folder `C:/Users/<username>/AppData/Local/Discord`

* The github installer still fails and the log clearly indicates the wrong installer is running still.

3)  deleted the folder `C:/Users/<username>/AppData/Local/SquirrelTemp`

 **At this point the github desktop installer works correctly**

4) redownload discord installer and run it

 **this also works without any monkey business**

TL;DR if you're having this issue go delete `C:/Users/<username>/AppData/Local/SquirrelTemp` and the github installer will likely work. It seems the discord installer is smart enough to work around the issue but the github installer is not.

**Comment by SamHunny** [comment]
As of April 2025, @espaulding 's suggestion was the only one that worked, except that I also had to uninstall Slack (also uses Squirrel) and reboot before GitHub would install.

**Comment by FSSZilla** [comment]
June 30, 2026, same issue as described above. I was able to install successfully after removing the SquirrelTemp folder as suggested in @espaulding 's steps above, even without uninstalling Discord.
